scripts/build-packages-dist.sh

```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/env bash
+
+set -u -e -o pipefail
+
+# Go to project directory.
+cd $(dirname ${0})/..
+
+# Either "legacy" (view engine) or "aot" (ivy)
+compileMode=${1:-"legacy"}
+
+# Path to the bazel binary. By default uses "bazel" from the node modules
+# but developers can overwrite the binary though an environment variable.
+bazelBin=${BAZEL_BIN_PATH:-$(yarn bin bazel)}
+
+echo "######################################"
+echo "  building release packages"
+echo "  mode: ${compileMode}"
+echo "######################################"
+echo ""
+
+# Path to the output directory into which we copy the npm packages.
+destPath="dist/releases"
+
+# Path to the bazel-bin directory.
+bazelBinPath=$(${bazelBin} info bazel-bin 2> /dev/null)
+
+# List of targets that need to be built, e.g. //src/lib, //src/cdk, etc. Note we need to remove all
+# carriage returns because Bazel prints these on Windows. This breaks the Bash array parsing.
+targets=$(${bazelBin} query --output=label 'attr("tags", "\[.*release-package.*\]", //src/...)' \
+  'intersect kind(".*_package", //src/...)' 2> /dev/null | tr -d "\r")
```

Comment:
We actually want to have the possibility to only select specific targets. (i.e. through a special tag). This is because we don't want to build `material-examples` for example. Without a special tag it could also happen that the query matches other nested packages such as `src/cdk/schematics:npm_package`.

Also we need to support both `npm_package` and `ng_package`.